### PR TITLE
Enable unit test to work on non-EN locales

### PR DESCRIPTION
**Pull request by offa**
The message may not be _"DOCTYPE is disallowed"_ depending on the locale, non English locales fail the test therefore. 
Reducing the assertion to the `DOCTYPE` keyword should work on each language.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@jtnord 

Since this is a security related test, some related reviewers may be good as well.

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by alecharp** [comment]
@offa Could you show us a time this test failed? I couldn't find a build on CI where this test failed. Thanks.

**Comment by offa** [comment]
There's no failing CI build since it runs on an english locale (I assume ...). Having some different language settings yields a translated exception message, not matching the english string of the test.

**Comment by alecharp** [comment]
I'm sorry, I have the same locale on my setup. Could you show us the string that you have, if you are using a different locale? Thank you for your time.

**Comment by offa** [comment]
I'm struggling to get tests use `-Duser.language=... -Duser.region=...`, so I've set them in the test. 

Here are some examples using locales defined in `Locales` (random pick):

| Locale | Message |
|----------|----------|
| Italian | DOCTYPE non è consentito se la funzione "http://apache.org/xml/features/disallow-doctype-decl" è impostata su true. |
| French | DOCTYPE n'est pas autorisé lorsque la fonctionnalité "http://apache.org/xml/features/disallow-doctype-decl" est définie sur True. |
| Japanese | 機能"http://apache.org/xml/features/disallow-doctype-decl"がtrueに設定されている場合、DOCTYPEは指定できません。 |
| German | DOCTYPE ist nicht zulässig, wenn das Feature "http://apache.org/xml/features/disallow-doctype-decl" auf "true" gesetzt ist. |
| Simplified Chinese | 将功能 "http://apache.org/xml/features/disallow-doctype-decl" 设置为“真”时, 不允许使用 DOCTYPE。 |

File: core/src/test/java/hudson/PluginManagerTest.java
Comment (basil) [inline core/src/test/java/hudson/PluginManagerTest.java:88]: Why not also check that the message contains the URL, as that is the same in all translations?

```suggestion
        assertThat(ex.getCause().getMessage(), containsString("DOCTYPE"));
        assertThat(ex.getCause().getMessage(), containsString("http://apache.org/xml/features/disallow-doctype-decl"));
```

**Review by basil — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]
Reasonable, I can replicate the test failure on a quickly spun up German localized environment.

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
